test/unit/gateways/first_pay_json_test.rb

```diff
@@ -0,0 +1,454 @@
+require 'test_helper'
+
+class FirstPayJsonTest < Test::Unit::TestCase
+  include CommStub
+
+  def setup
+    @gateway = FirstPayJsonGateway.new(
+      processor_id: 1234,
+      merchant_key: 'a91c38c3-7d7f-4d29-acc7-927b4dca0dbe'
+    )
+
+    @credit_card = credit_card
+    @amount = 100
+
+    @options = {
+      order_id: SecureRandom.hex(24),
+      billing_address: address
+    }
+  end
+
+  def test_successful_purchase
+    response = stub_comms do
+      @gateway.purchase(@amount, @credit_card, @options)
+    end.check_request do |_endpoint, data, _headers|
+      assert_match(/transactionAmount/, data)
```

Comment:
💬 Comment:

If the Request body is going to be a JSON string, the proper way to test the structure would be to parse it and check the key-values